### PR TITLE
Chore: StoreSplit tag service

### PR DESCRIPTION
**Pull request by ying-jeanne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/annotations/annotationsimpl/xorm_store.go [hunk 1]
```diff
@@ -41,40 +42,41 @@ func validateTimeRange(item *annotations.Item) error {
...
 			return err
 		}
 
 		if item.Tags != nil {
-			tags, err := sqlstore.EnsureTagsExist(sess, tags)
+			tags, err := r.tagService.EnsureTagsExist(ctx, tags)
 			if err != nil {
 				return err
 			}
+			spew.Dump("here is the annotation %v >>>>>>>>>>> %v", item)
```
Comment (mildwonkey) [inline pkg/services/annotations/annotationsimpl/xorm_store.go:73]: I think this can be removed now :)

**Comment by ying-jeanne** [comment]
I decided to not put the EnsureTagsExist into transaction, since:
the database blocked error is because WithTransactionalDbSession only read the transaction from context, if not exist, it create a new transaction without put it into context, to actually use transaction correctly in the context, we need to create session with InTransaction at all the top service level where `EnsureTagsExist` is called :meow_dizzy:. otherwise it is kind of useless
we do use get tags, and then if not exist we insert, if they are in the transaction, that means get could potentially not find any “not committed insert tag” and fail at commit, this is not a strong point tho.
i can still keep sqlstore.DBSession as an input parameter as before, then the transaction should work easily, but then store would have import cycle
The PR is just a frist round, since not service is sometime injected into store for example dashboardstore instead of service, but we will do it in the second round since too much refectory needed.

**Review by zserge — APPROVED** [review]

**Review by mildwonkey — APPROVED** [review]
LGTM - just take out that one `spew` please :)

**Comment by ying-jeanne** [comment]
> LGTM - just take out that one `spew` please :)

done ;)

**Pull request merged**